### PR TITLE
[Xwt,Gtk,WPF] Add support for NewText handling in TextCellView

### PR DESCRIPTION
**Pull request by Therzok**
I have not tested the WPF backend, but pretty sure the Gtk backend is right.

This adds support for checking whether or not to modify a text cell's text based on the text being set.

### CODE REVIEW

File: Xwt.WPF/Xwt.WPFBackend/TextEntryBackend.cs [hunk 1]
```diff
@@ -225,7 +225,7 @@ private void OnActivated(object sender, System.Windows.Input.KeyEventArgs e)
 				Context.InvokeUserCode (EventSink.OnActivated);
 		}
 
-		private void OnTextChanged (object s, TextChangedEventArgs e)
+		private void OnTextChanged (object s, System.Windows.Controls.TextChangedEventArgs e)
```
Comment (sevoku) [inline Xwt.WPF/Xwt.WPFBackend/TextEntryBackend.cs]: is this needed?

File: Xwt/Xwt/TextChangedEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿//
+// TextChangedEventArgs.cs
+//
+// Author:
+//       therzok <>
```
Comment (sevoku) [inline Xwt/Xwt/TextChangedEventArgs.cs]: add your name and copyright to the header

**Comment by sevoku** [comment]
WPF is tricky, it uses data bindings and the `TextChanged` event (https://github.com/mono/xwt/blob/ed5e1a30c535ddda82e661cd1033dff6d6d7769e/Xwt.WPF/Xwt.WPFBackend.CellViews/TextCellViewBackend.cs#L36) is raised after the data has been changed already. I tried to use a custom TextBox and override OnTextChanged, but it has the same problem (approach I used for CheckBox in https://github.com/mono/xwt/commit/8f23789090ea2bb54f1c962e0fcbdde512ccdbc1#diff-ddcae0ee4362cba10902c4ecfaf62894R66 doesn't work here). We handle `PreviewTextInput`, but it would need much more logic around it to handle backspace etc.

And does it build with XamMac?

File: Xwt.WPF/Xwt.WPFBackend/TextEntryBackend.cs (hunk 1 above)
Comment (Therzok) [inline Xwt.WPF/Xwt.WPFBackend/TextEntryBackend.cs]: Not sure.

File: Xwt/Xwt/TextChangedEventArgs.cs (hunk 2 above)
Comment (Therzok) [inline Xwt/Xwt/TextChangedEventArgs.cs]: Will fix.

**Comment by Therzok** [comment]
XamMac doesn't have this implemented at all at the moment, so yes, it builds.

Maybe we can use PreviewTextInput for this, but that's raised on any keypress, not just text changed. Maybe TextChangedEventArgs could have both `oldText` and `newText`, and we keep an internal state in PreviewTextInput and put both of them in the event args?

**Comment by Therzok** [comment]
About WPF, can't we cache each TextChanged call argument and provide the old value? I don't suppose the value can change without notification.

**Comment by sevoku** [comment]
@Therzok I've added a fix for WPF: 09bb47e
It simply uses the Text Binding to reset the text to its original value.

**Comment by Therzok** [comment]
Can we merge this and publish a new nuget? Can't wait to finish the resx editor.

**Pull request merged**